R/standardize_subject_ids.R

```diff
@@ -0,0 +1,213 @@
+#' Check whether the subject IDs comply with the expected format. When incorrect
+#' IDs are found, the function sends a warning and the user can call the
+#' `correct_subject_ids()` function to correct them.
+#'
+#' @param data The input data frame or linelist
+#' @param target_columns A vector of column names with the subject ids.
+#' @param prefix A prefix used in the subject IDs
+#' @param suffix A suffix used in the subject IDs
+#' @param range A vector with the range of numbers in the sample IDs
+#' @param nchar An integer that represents the expected number of characters in
+#'    the subject ids.
+#'
+#' @returns The input dataset with a warning if incorrect subject ids were found
+#'
+#' @examples
+#' dat <- check_subject_ids(
+#'   data           = readRDS(system.file("extdata", "test_df.RDS",
+#'                                        package = "cleanepi")),
+#'   target_columns = "study_id",
+#'   prefix         = "PS",
+#'   suffix         = "P2",
+#'   range          = c(1, 100),
+#'   nchar          = 7
+#' )
+#' @export
+check_subject_ids <- function(data,
+                              target_columns,
+                              prefix         = NULL,
+                              suffix         = NULL,
+                              range          = NULL,
+                              nchar          = NULL) {
+  checkmate::assert_data_frame(data, null.ok = FALSE)
+  checkmate::assert_character(target_columns, null.ok = FALSE,
+                              any.missing = FALSE, len = 1L)
+  checkmate::assert_character(prefix, len = 1L, null.ok = TRUE,
+                              any.missing = FALSE)
+  checkmate::assert_character(suffix, len = 1L, null.ok = TRUE,
+                              any.missing = FALSE)
+  checkmate::assert_vector(range, any.missing = FALSE, min.len = 2L,
+                           null.ok = TRUE, unique = TRUE, max.len = 2L)
+  checkmate::assert_numeric(nchar, null.ok = TRUE, any.missing = FALSE,
+                            len = 1L)
+
+  # coerce id column to character
+  if (is.numeric(data[[target_columns]]) | is.factor(data[[target_columns]])) {
```

Comment:
Should this be `||` or be wrapped in `any()`/etc.? At the moment, it can return a logical of length > 1 which will cause a problem in an `if`.

Reply:
It is supposed to be `||`.

fixed in commit `ee81659a`.